File: Data Structures & Algorithms/validate-parentheses/submission-6.py
class Solution:
    def isValid(self, s: str) -> bool:
        closeToOpen = {')':'(',']':'[','}':'{'}
        stack = []
        top = 0
        if len(s)<=1:
            return False
        for st in s:
            if st in closeToOpen:
                if(len(stack)==0):
                    return False
                # check top
                if(stack[top] == closeToOpen.get(st)):
                    stack.pop()
                    top = len(stack) - 1
                else:
                    return False
                
            if st=='(' or st=='[' or st=='{':
                stack.append(st)
                top = len(stack) - 1
        
        if len(stack) == 0:
            return True
        else:
            return False

File: Data Structures & Algorithms/minimum-stack/submission-6.java
class MinStack {

    Stack<Integer> stack = new Stack<>();
    Stack<Integer> minStack = new Stack<>();

    public MinStack() {
        
    }
    
    public void push(int val) {
        stack.push(val);

        if(minStack.isEmpty()){
            minStack.push(val);
        }

        else if(minStack.peek() > val){
            minStack.push(val);

        }
        else{
            minStack.push(minStack.peek());
        }
    }
    
    public void pop() {
        stack.pop();
        minStack.pop();
    }
    
    public int top() {
        return stack.peek();
    }
    
    public int getMin() {
        return minStack.peek();
    }
}
